BOY open detached sometimes doesn't open in databrowser perspective

BOY open detached sometimes doesn't open in databrowser perspective, I get the following error below.  If I change to OPI runtime perspective the viewpart opens fine.

!ENTRY org.eclipse.ui 4 0 2013-12-10 09:18:55.504
!MESSAGE Unhandled event loop exception
!STACK 0
org.eclipse.swt.SWTException: Failed to execute runnable (java.lang.ClassCastException: org.eclipse.ui.internal.ErrorViewPart cannot be cast to org.csstudio.opibuilder.runmode.IOPIRuntime)
    at org.eclipse.swt.SWT.error(SWT.java:4282)
    at org.eclipse.swt.SWT.error(SWT.java:4197)
    at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:138)
    at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4140)
    at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3757)
    at org.eclipse.ui.internal.Workbench.runEventLoop(Workbench.java:2701)
    at org.eclipse.ui.internal.Workbench.runUI(Workbench.java:2665)
    at org.eclipse.ui.internal.Workbench.access$4(Workbench.java:2499)
    at org.eclipse.ui.internal.Workbench$7.run(Workbench.java:679)
    at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:332)
    at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:668)
    at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:149)
    at org.csstudio.utility.product.Workbench.runWorkbench(Unknown Source)
    at org.csstudio.startup.application.Application.startApplication(Unknown Source)
    at org.csstudio.startup.application.Application.start(Unknown Source)
    at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:110)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:79)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:344)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:179)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
    at java.lang.reflect.Method.invoke(Unknown Source)
    at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:622)
    at org.eclipse.equinox.launcher.Main.basicRun(Main.java:577)
    at org.eclipse.equinox.launcher.Main.run(Main.java:1410)
    at org.eclipse.equinox.launcher.Main.main(Main.java:1386)
Caused by: java.lang.ClassCastException: org.eclipse.ui.internal.ErrorViewPart cannot be cast to org.csstudio.opibuilder.runmode.IOPIRuntime
    at org.csstudio.opibuilder.runmode.RunModeService$2.run(Unknown Source)
    at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:35)
    at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:135)
    ... 25 more

--- a/applications/plugins/org.csstudio.opibuilder/src/org/csstudio/opibuilder/runmode/RunModeService.java
+++ b/applications/plugins/org.csstudio.opibuilder/src/org/csstudio/opibuilder/runmode/RunModeService.java
@@ -191,19 +191,25 @@ public class RunModeService {
 	            final IWorkbenchWindow window = workbench.getActiveWorkbenchWindow();
 	            final IWorkbenchPage page = window.getActivePage();
 				try {
-					
-					
-					
 					IViewReference[] viewReferences = page.getViewReferences();
 					//If it is already opened.
 					for(IViewReference viewReference : viewReferences){
 						if(viewReference.getId().equals(OPIView.ID)){
-							if(runnerInput.equals(
-									((IOPIRuntime)viewReference.getView(true)).getOPIInput())){
-								page.showView(
-										OPIView.ID, viewReference.getSecondaryId(), IWorkbenchPage.VIEW_ACTIVATE);
-								return;								
-							}
+						    final IViewPart view = viewReference.getView(true);
+						    if (view instanceof IOPIRuntime)
+						    {
+						        final IOPIRuntime opi_view = (IOPIRuntime)view;
+    							if (runnerInput.equals(opi_view.getOPIInput()))
+    							{
+    								page.showView(OPIView.ID, viewReference.getSecondaryId(), IWorkbenchPage.VIEW_ACTIVATE);
+    								return;								
+    							}
+						    }
+						    else
+						    {
+					            OPIBuilderPlugin.getLogger().log(Level.WARNING,
+				                    "Found view " + view.getTitle() + " but its type is " + view.getClass().getName());
+						    }
 						}
 					}
 					

